src/elements/content-explorer/ContentExplorer.js

```diff
@@ -370,6 +380,23 @@ class ContentExplorer extends Component<Props, State> {
      */
     showMetadataQueryResults() {
         const { metadataQuery }: Props = this.props;
+        const { currentPageNumber, markers }: State = this.state;
+        const metadataQueryClone = cloneDeep(metadataQuery);
+
+        if (currentPageNumber === 0) {
+            // Preserve the marker as part of the original query
+            markers[currentPageNumber] = metadataQueryClone.marker;
```

Comment:
looks like markers is an array, but seems to be used like an object, might as well make it an object with number keys?

Reply:
@priyajeet `markers` is an array with `pageNumbers` used as index which maps nicely in this use case where if you land on page `x`, we query and populate the marker at index `x+1`. Any reason why use object with numeric keys than just sticking to array and indices?